tests/integration/web/auth/sudo_test.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+from django.test import RequestFactory
+from django.contrib.sessions.middleware import SessionMiddleware
+
+from nav.web.auth.utils import set_account
+from nav.web.auth.sudo import sudo, desudo
+from nav.models.profiles import Account
+
+
+def test_sudo_should_change_session_id(
+    db, session_request, admin_account, other_account
+):
+    # login with admin acount
+    set_account(session_request, admin_account)
+
+    pre_sudo_session_id = session_request.session.session_key
+    sudo(session_request, other_account)
+    post_sudo_session_id = session_request.session.session_key
+
+    assert pre_sudo_session_id != post_sudo_session_id
+
+
+def test_desudo_should_change_session_id(
+    db, session_request, admin_account, other_account
+):
+    # login with admin acount
+    set_account(session_request, admin_account)
+
+    sudo(session_request, other_account)
+
+    pre_desudo_session_id = session_request.session.session_key
+    desudo(session_request)
+    post_desudo_session_id = session_request.session.session_key
+
+    assert pre_desudo_session_id != post_desudo_session_id
```

Comment:
After desudo we should have the session id prior to the sudo. We should test for that.

Reply:
I think its better to just let the session after desudo be a new session, I think its safest to not reuse sessions like that. Found this from OWASP

https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#renew-the-session-id-after-any-privilege-level-change

> For all sensitive pages of the web application, any previous session IDs must be ignored, only the current session ID must be assigned to every new request received for the protected resource, and the old or previous session ID must be destroyed